contracts/Treasury.sol

```diff
@@ -6,48 +6,45 @@ import '@openzeppelin/contracts/utils/Address.sol';
 import '@openzeppelin/contracts/utils/Context.sol';
 import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
 import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import './interfaces/IGovernance.sol';
 import './interfaces/IRewards.sol';
 
 /// @title Treasury contract.
 contract Treasury is Context {
     using SafeERC20 for IERC20;
 
-    address public governance;
-    address public executor;
+    IGovernance public governance;
 
     event Received(address from, address asset, uint256 amount);
     event Sent(address to, address asset, uint256 amount);
     event IncreasedAllowance(address spender, address asset, uint256 amount);
     event DecreasedAllowance(address spender, address asset, uint256 amount);
 
     modifier onlyGovernance() {
-        require(_msgSender() == governance, 'Treasury: onlyGovernance');
-        _;
-    }
-
-    modifier onlyExecutor() {
-        require(_msgSender() == executor, 'Treasury: onlyExecutor');
+        require(
+            _msgSender() == governance.executor(),
```

Comment:
View functions don't cost anything additionally if called outside of the contract, so its just cost of reading storage, same for if compared to variable inside contract.

Reply:
Hmmm, I'm not sure that's quite right 🤔 

I threw together a rough gas compare test @ https://github.com/CjHare/gas-usage-read-address

```
Gas cost reading variables
read from this contract, gas used 28615
    ✓ read address variable stored in contract
read from another contract, gas used 31096
    ✓ read address from another contract
```

It's a artificial test, using an assignment to keep things as a transaction, to measure the gas usage within a transaction, rather then a view from offchain, but one number is bigger than the other.
(Maybe there is a fault in my test contracts or test 🤷 )

Reply:
Oooh, okay I just tried to google and everywhere in docs/responses I’ve seen that external/public view functions called externally dont use any more gas. I’ll also try myself and make the function executor() external instead of public to see the difference, public functions are in general as practice shows cost a bit more (at least if params involved).

Reply:
Ah, the joys of the internet!

View function can be gas zero, but less often when call from a contract compared to offchain (JSON-RPC)

```
View functions do not require any gas in order to be executed if :
It is called externally
It is called internally by another view function.
```
https://jeancvllr.medium.com/solidity-tutorial-all-about-functions-dba2ccb1e931

Reply:
Yeah, thats what we do here, calling externally.

Reply:
Externally usually refers to an off-chain call to the view function, although maybe the article bundles contract-contract with off-chain.

Anyway, when you already have a transaction initiated, anything done within the scope of that costs. Even internal memory operations cost gas 

Also means that linked Medium article is misleading as it claimed 'no gas' 🤔 ...which obviously can't be true, as every operation cost gas (even if only a little).

Reply:
I will put executor back then.

Reply:
Alright, I've actually found a decent article describing view functions within the context of a wider state changing transaction.
https://blog.b9lab.com/calls-vs-transactions-in-ethereum-smart-contracts-62d6b17d0bc2

_The B9Lab article touches on the various combinations you can have with the call operations and potential zero gas cost_